js_modules/dagit/packages/core/src/ticks/tests/DryRunRequestTable.test.tsx

```diff
@@ -0,0 +1,54 @@
+import {Resolvers} from '@apollo/client';
+import {MockedProvider, MockedResponse} from '@apollo/client/testing';
+import {act, getByText, queryAllByRole, render, screen, waitFor} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+import {BrowserRouter, Router} from 'react-router-dom';
+
+import {RunRequest} from '../../graphql/types';
+import {TestProvider} from '../../testing/TestProvider';
+import {RunRequestTable} from '../DryRunRequestTable';
+
+import {mockRepository} from './DryRunRequestTable.mocks';
+import {SensorDryRunMutationRunRequests} from './SensorDryRun.mocks';
+
+jest.mock('../../workspace/WorkspaceContext', () => ({useRepository: () => mockRepository}));
+
+// Not sure how to better handle this. Since we typed our mock with MockedResponse typescript thinks
+// these fields are nullable but they're not.
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const runRequests = SensorDryRunMutationRunRequests!.result!.data.sensorDryRun.evaluationResult
+  .runRequests as RunRequest[];
```

Comment:
Can you define the `runRequests` values separately (typed as `RunRequest[]`) and then make your mock use that? Then you can just reference them directly here and they should have the correct types since they're independent of the full `MockedResponse`?

```
const runRequests: RunRequest[] = [
  // etc.
];

// later
...
evaluationResult: runRequests,
...
```

Reply:
yeah, thats what I ended up doing in schedules PR, I'll copy that here too